### PR TITLE
feat: add quote_qty plumbing for quote-native MARKET BUY (v0.56.0)

### PR DESCRIPTION
**Pull request by blahmonkey**
**NOTE:** The author must always **first review the PR themselves**, before requesting review from others, that means carefully having reviewed the full diff in GitHub.

Companion to Praxis [`feat/quote-qty-market-buy`](https://github.com/Vaquum/Praxis/tree/feat/quote-qty-market-buy) — this PR ships the **Nexus side only**. The two PRs are paired: Nexus alone is a no-op (the new `Action.quote_qty` field threads through `translate_to_trade_command` and `PraxisOutbound.send_command`, but Praxis must accept `quote_qty=` on `submit_command` for any quote-native ENTER to actually leave Nexus).

## What does this PR change?

Replaces the v0.55.0 Praxis-side 200-bps reservation buffer (`order_notional = order_size * reference_price * Decimal('1.02')`) with Binance's native `quoteOrderQty` parameter. Strategies emit a quote-asset spend cap directly instead of pre-computing `size = notional / reference_price`; the venue determines the executed base quantity from live liquidity, and the reservation matches spend exactly — no kline-staleness slippage gap to compensate for.

### `feat:` `quote_qty: Decimal | None` field on `Action`

New field on [`Action`](nexus/strategy/action.py) with `__post_init__` validation: finite positive Decimal or `None`; mutually exclusive with `size` on ENTER (exactly one required); must be unset on EXIT. ENTER with `quote_qty` is restricted to `direction == OrderSide.BUY` and `order_type == OrderType.MARKET` so a malformed action is rejected at strategy boundary before any reservation is taken — Binance's `quoteOrderQty` REST parameter only accepts MARKET BUY.

### `feat:` `quote_qty: Decimal | None` field on `TradeCommand`

New field on [`TradeCommand`](nexus/infrastructure/praxis_connector/trade_command.py) with `__post_init__` validation: finite positive Decimal or `None`; on `NEW_ORDER` requires exactly one of `size` or `quote_qty`; must be unset on `AMEND_ORDER` and `CANCEL_ORDER`. [`translate_to_trade_command`](nexus/infrastructure/praxis_connector/translate.py) threads `action.quote_qty` into the command; when set, `TradeCommand.size` is `None` and the venue determines the executed base from live liquidity. [`PraxisOutbound.send_command`](nexus/infrastructure/praxis_connector/praxis_outbound.py) sends `quote_qty=command.quote_qty` to Praxis's `submit_command`.

### `fix:` Relax `make_reference_integrity_hook` ENTER size check

[`make_reference_integrity_hook`](nexus/core/validator/intake_stage.py) now accepts `order_size is None` when `order_notional > 0` (quote-native ENTER carries no base size at submission time, only a notional spend cap). The two failure modes remain enforced: explicit `order_size <= 0` still fails with `INTAKE_ENTER_SIZE_INVALID`, and `order_size is None` with non-positive `order_notional` fails with a distinct message identifying the quote-native shape requirement.

### `fix:` `OrderContext.order_size` becomes `Decimal | None`

[`OrderContext.order_size`](nexus/infrastructure/praxis_connector/order_context.py) is required for EXIT (the controller's `_clear_pending_exit` path reads it for cancellation/reject release) but optional for ENTER (the entry path uses `outcome.fill_size` from the venue rather than the pre-submit reservation). `__post_init__` checks `is_entry` type **first** so a non-bool slipping through cannot bypass the size-required-for-EXIT branch.

### `feat:` Strategy example switched to `quote_qty`

[`logreg_binary_evsfd`](examples/strategies/logreg_binary_evsfd.py) ENTER now returns `quote_qty = capital_available * _ENTER_CAPITAL_FRACTION` (a quote-asset spend cap directly) instead of `size = notional / reference_price`. The early-return-on-no-price path is removed from `_enter` because capital sizing is now in quote units; `reference_price` is still forwarded for slippage observation by Praxis but is no longer load-bearing for the order shape.

## Why this PR exists — observed evidence from prod

On Praxis prod (v0.72.0 monkeypatch, Jun 2026), 1658 MARKET BUY ENTERs failed `order_fill` with `fill_notional > remaining`:

| Metric | Value |
|---|---:|
| Unique command_ids failing once | 1658 |
| bps distribution of fill_notional/remaining excess | p10=2.4, p50=15.7, p90=48.7, p99=154, max=170 |
| Reference-price source | kline 15m close (up to ~15min stale at execution time) |

The v0.72.0 pre-snap fix tightened the reservation to exactly `snapped_qty * reference_price`, exposing TD-046 (MARKET ENTER reservation has no slippage buffer). The 200-bps Praxis-side hotfix at [`praxis/launcher.py:683`](https://github.com/Vaquum/Praxis/blob/main/praxis/launcher.py#L683) restored throughput but is a workaround, not a fix: it over-reserves capital by ~2% on every ENTER to absorb a staleness gap that shouldn't exist in the first place.

Option C (this PR + Praxis companion) closes the gap at the source. The strategy emits `quote_qty = budget`; Binance executes against live liquidity within the spend cap; the reservation matches `quote_qty` exactly; no reference-price multiplication, no buffer needed.

## Verification

- `ruff check nexus tests` — clean
- `pytest tests/` — **1465 passed, 17 skipped**
- 14 new tests added across `test_strategy_event_types.py` (7), `test_trade_command.py` (6), and `test_validator_intake_stage.py` (1)
- pr-prep Phase 1b LLM-attribution: PASS
- pr-prep Phase 4 Greybeard (final): **mediocre → fix-all applied** (1 VALID_FIX addressed: tightened `OrderContext.__post_init__` is_entry bool-ordering; test coverage added for all new validation surfaces)
- Round-3 codex audit on combined Nexus + Praxis diffs: 9 gaps surfaced, 7 patched, 2 accepted (fee reservation is honest; `max_position` is not active today)
- Round-4 codex audit on combined diffs: gaps 1–7 confirmed closed; WS-path terminal handling for quote-native flagged as a separate Praxis-side follow-up (out of scope for this Nexus-side PR)

## Scope

[`nexus/strategy/action.py`](nexus/strategy/action.py) (new `quote_qty` field + side/order_type guards), [`nexus/infrastructure/praxis_connector/trade_command.py`](nexus/infrastructure/praxis_connector/trade_command.py) (new field + NEW_ORDER XOR + AMEND/CANCEL rejection), [`nexus/infrastructure/praxis_connector/translate.py`](nexus/infrastructure/praxis_connector/translate.py) (thread `action.quote_qty` into command), [`nexus/infrastructure/praxis_connector/praxis_outbound.py`](nexus/infrastructure/praxis_connector/praxis_outbound.py) (send to Praxis), [`nexus/infrastructure/praxis_connector/order_context.py`](nexus/infrastructure/praxis_connector/order_context.py) (`order_size` optional for ENTER, is_entry ordering tightened), [`nexus/core/validator/intake_stage.py`](nexus/core/validator/intake_stage.py) (ENTER size check relaxation for quote-native), [`examples/strategies/logreg_binary_evsfd.py`](examples/strategies/logreg_binary_evsfd.py) (switch ENTER to `quote_qty`), [`tests/test_strategy_event_types.py`](tests/test_strategy_event_types.py) (7 new Action tests), [`tests/test_trade_command.py`](tests/test_trade_command.py) (6 new TradeCommand tests), [`tests/test_validator_intake_stage.py`](tests/test_validator_intake_stage.py) (1 new intake test), [`pyproject.toml`](pyproject.toml) (`0.55.0 → 0.56.0`), [`CHANGELOG.md`](CHANGELOG.md) (v0.56.0).

## Checklist

- [x] I have reviewed full diff in "Files changed"
- [x] I left no unnecessary files in the changes
- [x] I ran `python -m pytest` locally without errors (1465 passed, 17 skipped)
- [ ] I updated `/docs` (if behavior/API/config/user/etc changed)
- [x] I added and/or updated docstrings as per [Writing Docstrings](https://github.com/Vaquum/dev-docs/blob/main/src/Writing-Docstrings.md) (for any changed public functions/classes)
- [x] I updated CHANGELOG.md
- [x] I updated pyproject.toml
- [x] I added and/or updated tests (if behavior changed or new code paths added)
- [ ] I validated changes manually
- [x] I validated changes with LLM
- [x] I removed any extraneous examples/comments
- [ ] I linked issue to auto-close on merge (e.g., "Fixes #123") when applicable — paired with Praxis companion branch; no single issue captures Option C scope

### CODE REVIEW

File: examples/strategies/logreg_binary_evsfd.py [hunk 1]
```diff
@@ -99,36 +99,35 @@ def on_shutdown(
...
             Action(
                 action_type=ActionType.ENTER,
                 direction=OrderSide.BUY,
-                size=size,
+                quote_qty=quote_qty,
                 execution_mode=ExecutionMode.SINGLE_SHOT,
                 order_type=OrderType.MARKET,
                 deadline=_EXECUTION_DEADLINE_SECONDS,
-                reference_price=reference_price,
+                reference_price=self._reference_price(signal),
```
Comment (zero-bang) [inline examples/strategies/logreg_binary_evsfd.py:130]: This PR removed the `_enter` early-return on a `None` reference price, but the `_reference_price` docstring (lines 169-170) still claims:

> `_enter` short-circuits to no action when this returns None, so the launcher's poller-fallback path is not exercised.

That is no longer true — `reference_price=self._reference_price(signal)` now passes `None` straight into the `Action` (valid, since `reference_price` is optional and only used for slippage observation). Since this is a template strategy others copy, the stale claim is worth correcting: update the `_reference_price` docstring to note that a `None` result is now forwarded as `reference_price=None` rather than suppressing the ENTER.

File: nexus/infrastructure/praxis_connector/translate.py [hunk 2]
```diff
@@ -96,6 +96,13 @@ def translate_to_trade_command(
             if deadline is None:
                 deadline = _EXIT_DEFAULT_DEADLINE_SECONDS
 
+    if is_new_order:
+        size = None if action.quote_qty is not None else context.order_size
```
Comment (zero-bang) [inline nexus/infrastructure/praxis_connector/translate.py:100]: Non-blocking, forward-looking on the component seam: when `action.quote_qty` is set, the venue spends up to `quote_qty`, but the Nexus-side capital reservation is taken against `context.order_notional` (set by the Praxis-side `build_context`, not by this diff). Nothing here asserts `order_notional == quote_qty` for the quote-native path. If the companion ever derives `order_notional` from a stale reference price again (the exact failure this PR fixes), the reservation and the actual spend silently diverge and the overfill / over-park class of bug returns in a new form.

The predictable next failure mode: a quote-native ENTER where `build_context` computes `order_notional` from `quote_qty / reference_price * reference_price` with rounding/staleness drift, so the reservation no longer matches the spend cap. Worth a one-line invariant (or a paired test across the two repos) pinning `order_notional == quote_qty` for the quote-native shape — even if it lives Praxis-side.

**Review by zero-bang — COMMENTED** [review]
Reviewed the full diff and traced consumers across the seam. The `quote_qty` plumbing is clean and well-tested:

- `Action` / `TradeCommand` validation is thorough — NEW_ORDER XOR (`size`/`quote_qty`), BUY+MARKET restriction matching Binance's `quoteOrderQty` constraint, AMEND/CANCEL/EXIT rejection.
- `OrderContext.order_size` optionality is safe: every read of `context.order_size` in `outcome_processor.py` (lines 264, 292-293) is gated under `if context.is_exit:`, and the ENTER fill path (`_handle_fill`) uses `outcome.fill_size`, never the pre-submit reservation. EXIT still requires a finite positive size.
- The intake relaxation is correct and the downstream stages (CAPITAL, PLATFORM_LIMITS) key off `order_notional`, not `order_size`, so a `None` base size flows through without breaking the reservation path. PRICE-stage slippage protection still runs against the book snapshot independent of the (now-decorative) action `reference_price`.

No blocking issues. Two non-blocking notes inline — one stale docstring introduced by this PR, and one forward-looking seam observation.

For the record on broader engineering: the design correctly moves the size determination to the venue, which is the right fix for the kline-staleness gap. The one architectural thing to keep an eye on is the two-source-of-truth between `quote_qty` (venue spend cap) and `order_notional` (Nexus reservation) — see the translate.py note.

File: examples/strategies/logreg_binary_evsfd.py (hunk 1 above)
Comment (blahmonkey) [reply examples/strategies/logreg_binary_evsfd.py:130]: Addressed in 81552a7 (file `examples/strategies/logreg_binary_evsfd.py`). The `_reference_price` docstring is rewritten to describe the v0.56.0 contract: a `None` result is forwarded to `Action.reference_price` (observational only — the order is sized in quote units), and the helper also now sanitizes non-finite / non-positive Decimals to `None` so malformed signal payloads cannot raise in `Action.__post_init__`. Class sweep: the file has a single helper that returns `Decimal | None` from signal data (`_reference_price`); `size` is sourced from `context.capital_available` (internal), not the signal, so no other signal→Decimal ingress exists.

File: nexus/infrastructure/praxis_connector/translate.py (hunk 2 above)
Comment (blahmonkey) [reply nexus/infrastructure/praxis_connector/translate.py:100]: Acknowledged as a forward-looking, non-blocking comment. The `order_notional == quote_qty` invariant for the quote-native shape belongs on the Praxis side — `_build_enter_context` is where the derivation happens, and the companion `feat/quote-qty-market-buy` branch on Praxis already structures the quote-native arm so `order_notional = action.quote_qty` is the direct-assignment path (no reference-price multiplication). I will add an explicit `assert order_notional == action.quote_qty` plus a paired test on the Praxis PR; tracking it there keeps the invariant next to the code that can break it. No change in this PR.

File: nexus/infrastructure/praxis_connector/praxis_outbound.py [hunk 3]
```diff
@@ -83,6 +83,7 @@ def send_command(self, command: TradeCommand) -> str:
                 symbol=command.symbol,
                 side=command.side,
                 qty=command.size,
+                quote_qty=command.quote_qty,
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/infrastructure/praxis_connector/praxis_outbound.py]: Addressed in 81552a7 (file `nexus/infrastructure/praxis_connector/praxis_outbound.py`). `send_command` now builds `submit_kwargs` and includes `quote_qty` only when `command.quote_qty is not None`, so older Praxis `submit_command` signatures that pre-date this PR will not receive an unexpected-keyword `TypeError`. Two new tests pin both halves of the contract: `test_quote_qty_omitted_when_unset` and `test_quote_qty_passed_when_set` in `tests/test_praxis_outbound.py`. Class sweep: `submit_fn` is called from exactly one site in this module; no other forward-compat kwarg surfaces today, but the conditional-include rule applies to any future kwarg added before the Praxis side accepts it.

File: examples/strategies/logreg_binary_evsfd.py [hunk 4]
```diff
@@ -99,36 +99,35 @@ def on_shutdown(
...
             ),
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply examples/strategies/logreg_binary_evsfd.py:131]: Addressed in 81552a7 (file `examples/strategies/logreg_binary_evsfd.py`). `_reference_price` now returns `None` when the parsed Decimal is non-finite (`NaN`, `Infinity`) or non-positive (`<=0`), so a malformed `close` payload cannot reach `Action.__post_init__` and suppress the ENTER. The helper docstring was rewritten to describe the v0.56.0 contract (a `None` is forwarded to `Action.reference_price`, observational only).

Class sweep: `_reference_price` is the only signal→Decimal ingress in this strategy; `size` is sourced from `context.capital_available` (internal Nexus-side capital state), not the signal payload. So sanitizing in the helper covers the population. No per-example test file exists for this strategy (it is a template); the consumer-side invariants are pinned by `tests/test_strategy_event_types.py::TestAction::test_quote_qty_must_be_positive_decimal` and the existing `reference_price` rejection tests.

File: nexus/core/validator/intake_stage.py [hunk 5]
```diff
@@ -211,13 +211,26 @@ def hook(context: ValidationRequestContext) -> ValidationDecision | None:
...
                 decision = ValidationDecision(
                     allowed=False,
                     failed_stage=ValidationStage.INTAKE,
                     reason_code='INTAKE_ENTER_SIZE_INVALID',
                     message='ENTER requires order_size greater than zero',
                 )
+            elif context.order_size is None and (
+                context.order_notional is None
+                or context.order_notional <= _ZERO
+            ):
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/core/validator/intake_stage.py]: Addressed in 8330a8a (file `nexus/core/validator/intake_stage.py`). Dropped the entire `elif context.order_size is None and (...)` branch — it was dead code in two ways: (1) `context.order_notional is None` contradicts `ValidationRequestContext.order_notional: Decimal` (non-Optional, validated as finite non-negative in `__post_init__`), and (2) `context.order_notional <= _ZERO` is unreachable inside the hook because `intake_stage.py:404` already rejects ENTER with zero notional via `INTAKE_ORDER_NOTIONAL_ZERO` before hooks run at line 437. Quote-native ENTER (size=None, notional>0) now falls through to allowed, which is the correct behavior — `tests/test_validator_intake_stage.py::test_enter_quote_native_passes_with_none_size` still pins the happy path.

Class sweep: audited the full ref-hook ENTER arm against `ValidationRequestContext` invariants + the upstream built-in zero checks. The dropped branch was the only dead defense-in-depth introduced by v0.56.0. No other instances.

File: nexus/infrastructure/praxis_connector/order_context.py [hunk 6]
```diff
@@ -72,13 +75,25 @@ def __post_init__(self) -> None:
...
 
-        if self.order_size <= _ZERO:
-            msg = 'OrderContext.order_size must be positive'
-            raise ValueError(msg)
+        if self.order_size is None:
+
+            if not self.is_entry:
+                msg = 'OrderContext.order_size is required for EXIT orders'
+                raise ValueError(msg)
+
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/infrastructure/praxis_connector/order_context.py:87]: Addressed in 3f9fe7a (file `tests/test_order_context.py`). Added two tests under `TestOrderContextNumericValidation`:
- `test_entry_order_size_none_allowed` — `is_entry=True, order_size=None` constructs without error and `ctx.order_size is None`.
- `test_exit_order_size_none_rejected` — `is_entry=False, order_size=None` raises `ValueError` matching "order_size is required for EXIT orders".

Class sweep: `order_size` is the only field on `OrderContext` that became Optional in v0.56.0; the existing `TestOrderContextNumericValidation` already covers zero/negative/NaN/non-Decimal, and the two new tests close the None-branch population.

File: nexus/infrastructure/praxis_connector/translate.py [hunk 7]
```diff
@@ -96,6 +96,13 @@ def translate_to_trade_command(
...
+        quote_qty = action.quote_qty
+    else:
+        size = None
+        quote_qty = None
+
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/infrastructure/praxis_connector/translate.py:105]: Addressed in 3f9fe7a (file `tests/test_translate.py`). Added four tests:
- `test_quote_native_enter_translation` — quote-native ENTER `Action(quote_qty=...)` translates to a `NEW_ORDER` `TradeCommand` with `quote_qty=Decimal("100")` and `size is None`; `notional` matches `quote_qty`.
- `test_exit_translation_omits_quote_qty` — EXIT `NEW_ORDER` carries `size` and `quote_qty is None`.
- `test_modify_translation_omits_quote_qty` — `AMEND_ORDER` has both `size` and `quote_qty` as None.
- `test_cancel_translation_omits_quote_qty` — `CANCEL_ORDER` has both `size` and `quote_qty` as None.

Class sweep: only the NEW_ORDER ENTER arm of `translate_to_trade_command` does `quote_qty` branching (lines 99-104); AMEND/CANCEL set both `size` and `quote_qty` to None unconditionally per the TradeCommand invariants. The four tests cover that population. 1473 pytest pass (+6 net).

File: nexus/infrastructure/praxis_connector/trade_command.py [hunk 8]
```diff
@@ -31,7 +31,14 @@ class TradeCommand:
...
         side: BUY/SELL direction; None for AMEND_ORDER/CANCEL_ORDER.
-        size: Base asset quantity; None for AMEND_ORDER/CANCEL_ORDER.
+        size: Base asset quantity. Mutually exclusive with `quote_qty`
+            on NEW_ORDER (exactly one required). None for AMEND_ORDER /
+            CANCEL_ORDER.
+        quote_qty: Quote-asset spend (e.g. USDT) for quote-native
+            MARKET BUY NEW_ORDERs. The venue determines the executed
+            base quantity from live liquidity. Mutually exclusive with
+            `size` on NEW_ORDER. None for AMEND_ORDER / CANCEL_ORDER
+            and EXIT.
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/infrastructure/praxis_connector/trade_command.py]: Addressed in fadc09e (file `nexus/infrastructure/praxis_connector/trade_command.py`). Rephrased the `quote_qty` description to split the two enforcement layers explicitly: AMEND_ORDER / CANCEL_ORDER are type-enforced None by `__post_init__._validate_command_type_invariants`; EXIT NEW_ORDERs do not carry quote_qty either, but that is enforced upstream by `Action.__post_init__` (EXIT must not set quote_qty), with the translator forwarding the already-None Action field through. The TradeCommand layer itself has no enter/exit discriminator.

Class sweep: audited the full TradeCommand docstring against the type-invariant code path. Line 41 (the cited specimen) was the only place that overstated what the type enforces. Adjacent claims about `side` (line 33, type-enforced) and `trade_id` (line 43, descriptive context) are accurate.

File: tests/test_translate.py [hunk 9]
```diff
@@ -427,3 +427,93 @@ def test_missing_command_id_rejected() -> None:
...
+    assert cmd.quote_qty == Decimal('100')
+    assert cmd.size is None
+    assert cmd.notional == Decimal('100')
+
+
+def test_exit_translation_omits_quote_qty() -> None:
+    '''EXIT NEW_ORDER never carries `quote_qty` — the Action invariant
+    rejects it upstream, and the connector sets it explicitly to None
+    in the non-NEW_ORDER and exit branches.
+    '''
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply tests/test_translate.py]: Addressed in fadc09e (file `tests/test_translate.py`). Rewrote the `test_exit_translation_omits_quote_qty` docstring to describe the actual mechanism: EXIT actions go through the NEW_ORDER translation arm; the translator forwards `action.quote_qty` verbatim, and `Action.__post_init__` enforces `quote_qty is None` on EXIT, so the forwarded value is always None — no exit-specific branch in the translator exists. The assertions themselves were already correct.

Class sweep: reviewed every test docstring added in the previous round. The `test_modify_translation_omits_quote_qty` and `test_cancel_translation_omits_quote_qty` docstrings correctly attribute enforcement to "TradeCommand invariant" (those go through the `else` arm at translate.py:103-104 which sets `quote_qty = None` explicitly, plus the TradeCommand `__post_init__` rejects non-None on AMEND/CANCEL). `test_quote_native_enter_translation` docstring is accurate. Only the EXIT test docstring was wrong.

File: CHANGELOG.md [hunk 10]
```diff
@@ -670,3 +670,24 @@
...
+
+### Update
+
+- Switch [`logreg_binary_evsfd`](examples/strategies/logreg_binary_evsfd.py) ENTER from `size = notional / reference_price` to `quote_qty = capital_available * _ENTER_CAPITAL_FRACTION` — the strategy ships a quote-asset spend cap directly. No reference-price guard is required for sizing (capital is in quote units), so the early-return-on-no-price path is removed from `_enter`; `reference_price` is still forwarded for slippage observation by Praxis but is no longer load-bearing for the order shape
+- Relax ENTER intake at [`make_reference_integrity_hook`](nexus/core/validator/intake_stage.py) to accept `order_size is None` when `order_notional > 0` (quote-native ENTER carries no base size at submission time, only a notional spend cap). The two failure modes remain enforced: explicit `order_size <= 0` still fails with `INTAKE_ENTER_SIZE_INVALID`, and `order_size is None` with non-positive `order_notional` fails with a distinct message identifying the quote-native shape requirement
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply CHANGELOG.md]: Addressed in ed66779 (file `CHANGELOG.md`). Rewrote the v0.56.0 `Update / make_reference_integrity_hook` bullet to describe the actual two-layer enforcement: the built-in `INTAKE_ORDER_NOTIONAL_ZERO` check at [`intake_stage.py:404`](nexus/core/validator/intake_stage.py) rejects ENTER with `order_notional <= 0` before hooks run; the ref-hook then rejects explicit non-positive `order_size` with `INTAKE_ENTER_SIZE_INVALID`. Quote-native ENTER (`order_size is None`, `order_notional > 0`) falls through to allowed. The original bullet referenced a "distinct quote-native message" that was removed in round 2 (commit 8330a8a) when the redundant ref-hook branch was deleted as dead code.

Class sweep: audited every v0.56.0 bullet (NOTE, Add Action, Add TradeCommand, Update logreg_binary_evsfd, Update intake_stage, Update OrderContext, Fix) against current shipped code. Only the intake_stage bullet was stale.

**Review by zero-bang — APPROVED** [review]
